crates/solvers/src/infra/config/dex/balancer/file.rs

```diff
@@ -1,20 +1,17 @@
 use {
     crate::{
-        domain::{dex::slippage, eth},
-        infra::{contracts, dex},
-        util::conv,
+        domain::eth,
+        infra::{config::dex::file, contracts, dex},
     },
-    bigdecimal::BigDecimal,
     ethereum_types::H160,
     serde::Deserialize,
     serde_with::serde_as,
     std::path::Path,
-    tokio::fs,
 };
 
 #[serde_as]
 #[derive(Deserialize)]
-#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+#[serde(rename_all = "kebab-case")]
```

Comment:
I liked the `deny_unknown_fields` as it helps prevents misconfigurations (imagine you had an _optional_ field `foo` and you instead set `fou`, you might silently run thinking you have `foo` configured but in reality you do not).

You could instead flatten the fields here and have something like:
```rust
#[derive(Deserialize)]
#[serde(rename_all = "kebab-case", deny_unknown_fields)]
struct balancer::Config {
    // Balancer related fields.
    endpoint: reqwest::Url,
    vault: Option<H160>,
    settlement: Option<H160>,
    
    #[flatten]
    solver: infra::config::dex::solver::File,
}
```

And then `impl From<infra::config::dex::solver::File> for <infra::config::dex::solver::Config>`.

The unfortunate thing is that it leaks the `File` type, but you can scope that with `pub(super)` to limit it to the `infra::config::dex`.

Reply:
Alternatively, we can add nesting and then we don't need to flatten (and would essentially just change the format).

So:
```toml
endpoint = "https://my.endpoint"

[solver]
relative-slippage = "1.2"
```

Reply:
Nice, I like how it turned out.